### PR TITLE
WT-2222 Add statistics for named snapshots.

### PR DESCRIPTION
**Pull request by agorrod**
New statistics are:
* pinned by named snapshots
* number of named snapshots created
* number of named snapshots dropped

### CODE REVIEW

**Comment by agorrod** [comment]
@michaelcahill Would you mind reviewing this change?

File: dist/stat_data.py [hunk 1]
```diff
@@ -311,6 +311,13 @@ def __init__(self, name, desc, flags=''):
             'no_clear,no_scale'),
     TxnStat('txn_pinned_range',
         'transaction range of IDs currently pinned', 'no_clear,no_scale'),
+    TxnStat('txn_pinned_snapshot_range',
+        'transaction range of IDs currently pinned by named snapshots',
+            'no_clear,no_scale'),
+    TxnStat('txn_snapshots_created',
+        'number of named snapshots created', 'no_clear,no_scale'),
+    TxnStat('txn_snapshots_dropped',
+        'number of named snapshots dropped', 'no_clear,no_scale'),
```
Comment (michaelcahill) [inline dist/stat_data.py]: I didn't expect these counts of operations to be "no_clear,no_scale" -- was that intentional?

File: dist/stat_data.py [hunk 2]
```diff
@@ -311,6 +311,13 @@ def __init__(self, name, desc, flags=''):
             'no_clear,no_scale'),
     TxnStat('txn_pinned_range',
         'transaction range of IDs currently pinned', 'no_clear,no_scale'),
+    TxnStat('txn_pinned_snapshot_range',
+        'transaction range of IDs currently pinned by named snapshots',
+            'no_clear,no_scale'),
```
Comment (michaelcahill) [inline dist/stat_data.py]: To avoid confusion, "no_clear,no_scale" on the pinned range *does* look right.  Just not the on the following two.

**Comment by michaelcahill** [comment]
LGTM other than the "no_clear,no_scale" comment and with a merge of develop.

**Comment by michaelcahill** [comment]
Thanks, @agorrod, lgtm.  I'll merge.

**Pull request merged**